lib/versions/npm.js

```diff
@@ -1,82 +1,57 @@
-    }
-
-    this.logger.info(__("Downloading {{packageName}} {{version}}...", {packageName: packageName, version: version}));
+  static getPackagePath(packageName, version){
+    return './.embark/versions/' + packageName + '/' + version + '/' + packageName;
+  }
 
-    const obsMeasure = new PerformanceObserver((items) => {
-      let entry;
-      let strDuration;
+  _isInstalling(packageName, version){
+    return typeof this._installing[packageName + version] != 'undefined';
+  }
 
-      // find any download ongoing measurements we've made
-      entry = _.last(_.where(items.getEntries(), {name: downloadOngoing}));
-      if(entry){
-        // ongoing performance mark 
-        strDuration = __('Still downloading {{packageName}} {{version}}... ({{duration}}ms elapsed)', {packageName: packageName, version: version, duration: entry.duration});
-      }
-      else{
-        // otherwise, find our download complete measurement
-        entry = _.last(_.where(items.getEntries(), {name: downloadComplete}));
-        
-        if(entry){
-          strDuration = __('Finished downloading {{packageName}} {{version}} in {{duration}}ms', {packageName: packageName, version: version, duration: entry.duration});
-          performance.clearMarks();
-        }
-      }
-      
-      // log our measurement and make it red if it has taken too long
-      if(entry && strDuration){
-        if(entry.duration > 4000){
-          strDuration = strDuration.red;
-        }
-        this.logger.info(strDuration);
-      }
-      
-    });
-    obsMeasure.observe({entryTypes: ['measure']});
+  getPackageVersion(packageName, version, callback) {
+    const packagePath = Npm.getPackagePath(packageName, version);
 
-    // define mark and measurement names
-    let startMark = 'downloadStart' + packageName + version;
-    let ongoingMark = 'downloadOngoingMark' + packageName + version;
-    let downloadOngoing = 'downloadOngoing' + packageName + version;
-    let endMark = 'downloadEnd' + packageName + version;
-    let downloadComplete = 'downloadComplete' + packageName + version;
+    // check if this package already exists in the filesystem
+    if (fs.existsSync(packagePath)) {
+      return callback(null, packagePath);
+    }
 
-    // mark our start time
-    performance.mark(startMark);
+    const pluginManager = new PluginManager({pluginsPath: './.embark/versions/' + packageName + '/' + version + '/'});
     
-    // function that continually updates the console to show user that we're downloading a library
-    let intOngoingDownload = setInterval(
-      function(){ 
-        performance.mark(ongoingMark);
-        performance.measure(downloadOngoing, startMark, ongoingMark); 
-      }, 750);
+    // check if we're already installing this package
+    if(this._isInstalling(packageName, version)){
+      this._installing[packageName + version].push(callback);
+    }else{
+      this._installing[packageName + version] = new Array(callback);
```

Comment:
Just an FYI, you can actually just do `= [callback]`

Reply:
Fixed